File: Tools/TiCloudAccessToken/go/src/AccessTokenBuilder/AccessTokenBuilder.go
package AccessTokenBuilder

import (
	accesstoken "github.com/ti-net/Tools/TiCloudDynamicKey/go/src/AccessToken"
)

//AccessTokenBuilder class
type AccessTokenBuilder struct {
}

//BuildAccessTokenWithPrivilegeTs method
// enterpriseId: The enterpriseId issued to you by TiCloud.
// token:	The token issued to you by TiCloud.
// userId: The userId. optionalUid must be unique.
// expireTimestamp: represented by the number of seconds elapsed since 1/1/1970.
func BuildAccessTokenWithPrivilegeTs(enterpriseId string, token string, userId string, expireTimestamp uint32) (string, error) {
	accessToken := accesstoken.CreateAccessToken(enterpriseId, token, userId)
	accessToken.AddPrivilege(accesstoken.API, expireTimestamp)
	return accessToken.Build()
}

File: Tools/TiCloudAccessToken/php/src/AccessTokenBuilder.php
<?php

require_once "AccessToken.php";

class AccessTokenBuilder
{

    # enterpriseId: The enterpriseId issued to you by TiCloud.
    # token:	The token issued to you by TiCloud.
    # userId: The userId. optionalUid must be unique.
    # expires: expire seconds.
    public static function buildAccessToken($enterpriseId, $token, $userId, $expires)
    {
        $accessToken = AccessToken::init($enterpriseId, $token, $userId);
        $Privileges = AccessToken::Privileges;

        $currentTimestamp = (new DateTime("now", new DateTimeZone('UTC')))->getTimestamp();
        $expireTimestamp = $currentTimestamp + $expires;
        $accessToken->addPrivilege($Privileges["API"], $expireTimestamp);

        return $accessToken->build();
    }
}


?>
